### ISSUE TITLE
Trigger same md-menu from various contexts

### ISSUE THREAD

**Issue by acondiff** [angular/material#6148]
I would like to trigger `md-menu` by clicking buttons with `ng-click="myMenu.open($event)"` so my app can be more performant. I currently have a `ng-repeat` on hundreds of items with an `md-menu` in each. It is taking a while to render because of the `md-menu` directive running on all 100+ instances. It is all the same menu, I just need to pass in a variable to give it context of which item I clicked. 

I tried to do the following but got an odd error when doing so.

```
$rootScope.EntryMenuCtrl = {
  open: function(event) {
    console.log($rootScope.EntryMenuCtrl);
    $mdMenu.show({
      scope: $rootScope.$new(),
      mdMenuCtrl: $rootScope.EntryMenuCtrl,
      element: $templateCache.get('stream/entry-menu.html'),
      target: event.target, // used for where the menu animates out of
    });
  },
  close: function() { $mdMenu.hide(); },
  positionMode: function() { return {left: 'target', top: 'target'}; },
  offsets: function() { return {top: 0, left: 0,}; },
};
```

I get the following error:

```
TypeError: o.controller is not a function
    at Function.o.inherit (vendor-0.0.0.min.js:32457)
    at o (vendor-0.0.0.min.js:32502)
    at n.then.g (vendor-0.0.0.min.js:31033)
    at c (vendor-0.0.0.min.js:13248)
    at vendor-0.0.0.min.js:13264
    at m.$eval (vendor-0.0.0.min.js:14466)
    at m.$digest (vendor-0.0.0.min.js:14282)
    at m.scopePrototype.$digest (hint.js:1972)
    at m.$apply (vendor-0.0.0.min.js:14571)
    at m.scopePrototype.$apply (hint.js:2035)
```

It looks like it is being caused by the theming service oddly enough: 
```
  function ThemingService($rootScope, $log) {

    applyTheme.inherit = function(el, parent) {
      var ctrl = parent.controller('mdTheme');
      [...]
```

TL;DR: I need a way to open an `md-menu` I can declare once from many different contexts rather than having many unnecessary declarations of the same code slowing down my app.

**Comment by acondiff** [comment]
This my go along with this issue: https://github.com/angular/material/issues/3493

I'd say leave the right-click logic up to the developer, solving my problem may also solve this other issue as well.
